[Saezenn] iP

Saezenn opened the pull request:
# Sae 

> Sae is the name of the loveliest person you'll ever meet - Me


Sae frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x]  Managing tasks
- [x]  Managing deadlines
- [ ]  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
``` 
public static void main(String[] args) {
        Sae sae = new Sae("./data/sae.txt");
        sae.run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Saezenn committed:
Level-0, Rename, Greet, Exit

Saezenn committed:
Level-1, Echo

Saezenn committed:
Level-2, Add, List

Saezenn committed:
Level-3, Mark as Done

Saezenn committed:
Level-4, Todo, Event, Deadline

Saezenn committed:
Level-5, Handle Erorrs

Saezenn committed:
Level-6, Delete

Saezenn committed:
A-TextUiTesting: Automated Text UI Testing

Saezenn committed:
Fixed Exceptions, Text-Ui-Testing

Saezenn committed:
Level-7, Save

Saezenn committed:
Level-8, Dates and Times

Saezenn committed:
merging branch-Level-8 to master

Saezenn committed:
A-MoreOOP, TaskList, Storage, WIP

Saezenn committed:
Merge gradle-support to main

jedkohjk commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,81 @@
+        }
+    }
+
+    public void addtoStore() throws IOException {
```
```suggestion
    public void addToStore() throws IOException {
```

jedkohjk commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,81 @@
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
```
Is this out of lexicographical order?

jedkohjk commented on src/main/java/Sae.java:
```diff
@@ -0,0 +1,143 @@
+ */
+public class Sae {
+
+    private final Storage storage;
```
Should this be in capital letters?

jedkohjk commented on src/main/java/Event.java:
LGTM

jedkohjk commented on src/main/java/Sae.java:
```diff
@@ -0,0 +1,143 @@
+import java.io.IOException;
```
Are these in lexicographical order?

jedkohjk commented on src/main/java/InvalidDateTimeFormatException.java:
LGTM

jedkohjk commented on src/main/java/InvalidDeadlineException.java:
LGTM

jedkohjk commented on src/main/java/InvalidEventException.java:
LGTM

jedkohjk commented on src/main/java/InvalidTodoException.java:
LGTM

jedkohjk commented on src/main/java/SaeException.java:
LGTM

jedkohjk commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,90 @@
+import java.io.FileNotFoundException;
```
Are these in lexicographical order?

jedkohjk commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,90 @@
+                    newEvent.isDone = completion.equals("1");
+                    taskList.add(newEvent);
+                    break;
+            }
```
Is there a default case?

jedkohjk commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,77 @@
+/**
+ * The Task class represents a task with a description and a status indicating whether it's done or not.
+ */
+abstract public class Task {
```
```suggestion
public abstract class Task {
```

jedkohjk commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,77 @@
+
+    public abstract void addtoStore() throws IOException;
+
+    public void changeMarktoStore(String[] commandTask) {
```
```suggestion
    public void changeMarkToStore(String[] commandTask) {
```

jedkohjk commented on src/main/java/Todo.java:
```diff
@@ -0,0 +1,57 @@
+
+    }
+
+    public void addtoStore() throws IOException {
```
```suggestion
    public void addToStore() throws IOException {
```

jedkohjk reviewed (commented):
Overall, LGTM. Just a few nits on the import order.

Saezenn committed:
a-gradle

dloh2236 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,62 @@
+        System.out.println("Now you have " + store.size() + " tasks in the list.");
+    }
+
+    public void addtoStore() throws IOException {
```
should be addToStore

dloh2236 reviewed (commented):
minimal, see other commenters comments

Saezenn committed:
small updates

Saezenn committed:
Fixed OOP

Saezenn committed:
added packages

Saezenn committed:
Added JUnit Tests

Saezenn committed:
Create a JAR file

Saezenn committed:
Add javadocs to methods and classes

Saezenn committed:
Fix coding standard

Saezenn committed:
Add Level 9

Saezenn committed:
Add GUI

Saezenn committed:
Add assertions to code
Code does not have any assert statements.
Add assert statements to the code to define assumptions
Added to allow the runtime to verify assumptions and any violations.

Saezenn committed:
Merge pull request #2 from Saezenn/branch-A-Assertions

Add assertions to code

Code lacks assert statements.

Add assert statements to define assumptions for runtime verification.

The code does not currently include any assert statements, which makes it
challenging to verify assumptions during runtime. To address this issue, this
commit adds assert statements to the code. By doing so, we can ensure that
critical assumptions are validated, helping to prevent potential issues.

This change is made to enhance the code's reliability and maintainability during
development and debugging.

Saezenn committed:
Fix code quality

The current code lacks consistent formatting and import statement order.
This makes it harder to read and maintain.

Reformatted import statements alphabetically and fixed code formatting
according to Java conventions.

Prioritizing readability and maintainability to make the codebase more
organized and comprehensible.

Saezenn committed:
Merge pull request #3 from Saezenn/branch-A-CodeQuality

Fix code quality

The current code lacks consistent formatting and import statement order.
This makes it harder to read and maintain.

Reformatted import statements alphabetically and fixed code formatting
according to Java conventions.

Prioritizing readability and maintainability to make the codebase more
organized and comprehensible.

Saezenn committed:
Add C-MassOps

The application lacks the capability to efficiently delete a group of tasks
when needed.

Added the ability to mass delete a group of tasks by specifying a range
of task indices in the format 'delete 1 to 3.'

This feature enhances user experience and provides better task management
options.

Saezenn committed:
Merge pull request #4 from Saezenn/branch-C-MassOPs

Add C-MassOps

The application lacks the capability to efficiently delete a group of tasks
when needed.

Added the ability to mass delete a group of tasks by specifying a range
of task indices in the format "delete 1 to 3."

This feature enhances user experience and provides better task management
options.

Saezenn committed:
Add missing JavaDocs to classes and methods

Saezenn committed:
Merge pull request #5 from Saezenn/branch-A-JavaDoc

Add missing JavaDocs to classes and methods

Saezenn committed:
Fix some bugs in Storage

Saezenn committed:
Add Ui representative screenshot

Saezenn committed:
Complete A-UserGuide

Saezenn committed:
Edit A-UserGuide

Saezenn committed:
Fix issue when list called on empty tasklist

Saezenn committed:
Fix find method in TaskList

Saezenn committed:
Merge pull request #6 from Saezenn/Level-9

Fix find method in TaskList

Saezenn committed:
Fix javadocs and add new exceptions

Saezenn committed:
Merge pull request #7 from Saezenn/branch-A-CodingStandard

Fix javadocs and add new exceptions

Saezenn committed:
Fix javadocs

Saezenn committed:
Fix javadocs
